Chapter 6 Copyedit

carols10cents opened the pull request:
Changes for your approval @steveklabnik, I've taken a look at it and made a few further tweaks.

carols10cents commented:
Basically just look at https://github.com/rust-lang/book/pull/425/commits/5cba0cf6198fd4329b68bf83a659134312ad9d49

carols10cents commented on nostarch/chapter06.md:
```diff
@@ -44,25 +44,10 @@ enum IpAddrKind {
-Each variant is defined like a struct, and you instantiate both using the same
-syntax.
-
-<!-- end options -->
```
I guess they didn't like any of these options, after all that.

carols10cents commented on nostarch/chapter06.md:
```diff
@@ -668,63 +662,68 @@ match some_u8_value {
 
-We want to do something with the `Some` match, but do nothing with the `None`
-case. We can do this with an `Option`, but with a more complex enum,
-adding `_ => ()` after processing just one variant is a lot of boilerplate code
```
Ummm I might have changed the meaning of this paragraph a bit, rereading my rereading. It's kind of confusing, in any case... maybe we should use the 1, 3, 5 etc example again here instead...

steveklabnik replied:
Whatever you think is good 😄

steveklabnik commented on nostarch/chapter06.md:
```diff
@@ -150,19 +100,17 @@ let loopback = IpAddr {
 
 <caption>
-Listing 6-1: Storing the data and type of an IP address using a `struct`
+Listing 6-1: Storing the data and variety of an IP address using a `struct`
```
hrmmmmmmmmm

steveklabnik replied:
"variety" feels very weird to me here. idk.

carols10cents replied:
Yeah between all the uses of "type", "kind", etc everything was getting confusing, so I started subbing in nontechnical words when we were talking colloquially. What do you think about "Storing the data and `IpAddrKind` variant of an IP address" instead?

steveklabnik commented on nostarch/chapter06.md:
```diff
@@ -396,80 +340,71 @@ let sum = x + y;
-a similar way as the counting machine, determine which coin it is and
-return its value in cents, shown here in Listing 6-2:
+Because we just mentioned coins, let’s use them as an example using `match`! We
+can write a function that can take an unknown American coin and, in a similar
```
Could we say "US" instead of "American" here? Sorry I didn't catch that earlier.

Shout-out to my friends from South America.

steveklabnik commented on nostarch/chapter06.md:
```diff
@@ -549,12 +477,12 @@ Another useful feature of match arms is that they can bind to parts of the
-by changing the `Quarter` variant to include a `State` value stored inside it
-as we've done here in Listing 6-3:
+As an example, let’s change one of our enum variants to hold data inside it.
+From 1999 through 2008, the United States printed quarters with different
```
that will also end up making it more consistent here 😄

steveklabnik commented on nostarch/chapter06.md:
```diff
@@ -608,20 +536,21 @@ that point, the binding for `state` will be the value `UsState::Alaska`. We can
 state value out of the `Coin` enum variant for `Quarter`.
 
-### Matching with Option<T>
+### Matching with `Option`
```
hrm, officially, you're supposed to use `Option<T>`, and we do in most places.

steveklabnik commented on nostarch/chapter06.md:
```diff
@@ -608,20 +536,21 @@ that point, the binding for `state` will be the value `UsState::Alaska`. We can
-case when using `Option<T>`; we can do this in a very similar way! Instead of
-comparing coins we will be comparing the variants of `Option<T>`, but the way
-that the `match` expression works remains the same.
+case when using `Option<T>`; we can also handle `Option` using `match` as we
```
like, down here we have both in one line! idk.

steveklabnik commented on nostarch/chapter06.md:
```diff
@@ -734,63 +662,68 @@ match some_u8_value {
-`if let` takes a pattern and an expression separated by an `=`. It works
-just like a `match`, where the expression is given to the `match` and the
-pattern is its first arm.
+`if let` takes a pattern and an expression separated by an `=`. It works just
```
could we remove this just? in general i try not to use just

carols10cents commented:
Ok, I think I fixed all the things you mentioned, and fixed a few more things, in https://github.com/rust-lang/book/pull/425/commits/b69b4bd44bb16703dbd5bbfc154464ccddb41df5

Namely, I remembered to check if we have open issues for this chapter, and we do, #280. I decided to add one sentence listing some things patterns could be and point to the chapter that'll be all about patterns.

I also back/forward (idk which direction anything is anymore) propagated some changes we had made to the src/md since last time we had this chapter in https://github.com/rust-lang/book/pull/425/commits/d1aa4d806edcb08243ab9a3e6d36e23093f8d7ef, you've probably seen those changes already but just in case, you might want to take a peek. Up to you.

Wdyt?

steveklabnik commented:
LGTM. :shipit:
